src/components/ErrorBoundary/index.jsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+
+class ErrorBoundary extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = { hasError: false, error: null, errorInfo: null };
+    }
+
+    static getDerivedStateFromError(error) {
+        return { hasError: true };
+    }
+    
+    componentDidCatch(error, errorInfo) {
+        this.setState({
+            error: error,
+            errorInfo: errorInfo
+        })
+    }
+    
+    render() {
+        if (this.state.hasError) {
```

Comment:
Also you can use spread operator to reduce code a little:
```js
const {hasError, error} = this.state;
```

Reply:
Changed